### PR TITLE
misc: Allow setting init_containers with null condition

### PR DESCRIPTION
**Pull request by adamantike**
## what

This change allows configuring `init_containers` that are not added to the `depends_on` list for the main container, by setting `null` as the container `condition`.

## why

This is useful for containers that actually depend on the main container, and not the other way around. For example, a `nginx` sidecar that needs to wait for the main container to be healthy before starting.

By allowing `null` as the container condition, this is a non-breaking change.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -136,6 +136,7 @@ locals {
       containerName = lookup(jsondecode(init_container.container_definition), "name"),
```
[the remark this answers is not in the document]
Comment (adamantike) [reply main.tf:136]: Thanks for the quick review!

I notice that there are more linting issues than just that line, but the CI check seems to only be validating modified lines. I will open a separate PR to fix all linting issues, and we can come back to this one after `main` is error-free.

[the remark this answers is not in the document]
Comment (gberenice) [reply main.tf:136]: Sounds great, thank you!

[the remark this answers is not in the document]
Comment (adamantike) [reply main.tf]: Just rebased this PR on top of the latest commits!

**Review by gberenice — APPROVED** [review]

**Pull request merged**